### PR TITLE
Update to golang 1.18.6

### PR DESCRIPTION
**Pull request by aaronfern**
**What this PR does / why we need it**:
This PR updated the golang version used to build images from `1.18.3` to `1.18.6`
This PR also updated the pipeline definitions to use go `1.18.6`

**Which issue(s) this PR fixes**:
Fixes [CVE-2022-27664](https://nvd.nist.gov/vuln/detail/CVE-2022-27664)

**Special notes for your reviewer**:

**Release note**:

```noteworthy operator
Updated golang version used to build images to 1.18.6
```
```noteworthy operator
Updated golang version to 1.18
```

### CODE REVIEW

**Comment by ishan16696** [comment]
In druid we have updated the golang version to `1.18.5`.
I would suggest can we have a cosistent golang version in pipeline as well as to build images.

**Comment by aaronfern** [comment]
We need golang `1.18.6` here to solve this vulnerability
I will open a separate PR on etcd-druid to update the golang version used there to `1.18.6`

**Comment by ishan16696** [comment]
ok, make sense

File: .ci/pipeline_definitions [hunk 1]
```diff
@@ -42,13 +42,13 @@ etcd-backup-restore:
                 build: ~
     steps:
       check:
-        image: 'golang:1.17'
+        image: 'golang:1.18'
```
Comment (ishan16696) [inline .ci/pipeline_definitions:45]: Can you update pipeline to use `1.18.6` as druid pipeline is also at `1.1.8.5`

**Review by ishan16696 — CHANGES_REQUESTED** [review]

File: .ci/pipeline_definitions (hunk 1 above)
Comment (aaronfern) [reply .ci/pipeline_definitions]: Updated

**Review by ishan16696 — APPROVED** [review]

**Pull request merged**